onboarding_auth/onboarding_auth_presentation/src/main/java/com/alexafit/onboarding_auth_presentation/login/LoginScreen.kt

```diff
@@ -3,26 +3,107 @@ package com.alexafit.onboarding_auth_presentation.login
 import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material.MaterialTheme
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextAlign
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.alexafit.core.util.UiEvent
 import com.alexafit.core_ui.LocalSpacing
+import com.alexafit.core_ui.components.buttons.TextActionButton
+import com.alexafit.core_ui.components.textfield.TextField
+import com.alexafit.core_ui.R
+import com.alexafit.onboarding_auth_presentation.login.event.LoginEvent
 
 @Composable
 fun LoginScreen(
-    onNextClick: () -> Unit,
-    registerUser: () -> Unit
+    scaffoldState: ScaffoldState,
+    onNextClick: (LoginEvent) -> Unit,
```

Comment:
Why are you passing the Event on next?

If you are handling events, it would make more sense to name it something like
onEvent